src/test/ui/suggestions/return-bindings.rs

```diff
@@ -0,0 +1,15 @@
+fn a(i: i32) -> i32 {}
+//~^ ERROR mismatched types
+
+fn b(i: i32, j: i32) -> i32 {}
+//~^ ERROR mismatched types
+
+fn c(opt_str: Option<String>) {
+    let s: String = if let Some(s) = opt_str {
+        //~^ ERROR mismatched types
+    } else {
+        String::new()
+    };
+}
```

Comment:
What's the output for the following?

```
fn d(opt_str: Option<String>) {
    let s = if let Some(s) = opt_str {
        //~^ ERROR mismatched types
    } else {
        String::new()
    };
}
```

Should we also add it to the test?

Also, can this test be `// run-rustfix`?

Reply:
I'll check it out.

Reply:
```
error[E0308]: `if` and `else` have incompatible types
 --> /home/gh-compiler-errors/test.rs:5:9
  |
2 |       let s = if let Some(s) = opt_str {
  |  ______________________________________-
3 | |         //~^ ERROR mismatched types
4 | |     } else {
  | |_____- expected because of this
5 |           String::new()
  |           ^^^^^^^^^^^^^ expected `()`, found struct `String`
```

Reply:
It'd be nice if the suggestion worked there as well, but understand not tackling it now :)

Reply:
Filed https://github.com/rust-lang/rust/issues/99525